coreax/score_matching.py

```diff
@@ -545,7 +551,13 @@ def score_function(x_: Shaped[Array, " n d"]) -> Shaped[Array, " n d"]:
             :param x_: The :math:`n \times d` data vectors we wish to evaluate the score
                 function at
             """
-            # Check format
+            # Check format of input array. We use atleast_2d from JAX to perform
+            # conversions here. This is faster than using the custom function
+            # _atleast_2d_consistent in coreax.data, and whilst these two functions
+            # differ in output for 1-dimensional arrays, the computations inside of this
+            # function will match even in this case. This speed difference is noticeable
```

Comment:
> the computations inside this function will match

This isn't actually correct - running
```python
import jax.numpy as jnp

from coreax import KernelDensityMatching

matcher = KernelDensityMatching(length_scale=1)
data = jnp.array([[1, 2], [3, 4], [5, 6], [7, 8]])

fn = matcher.match(data)
print(fn(jnp.array([1, 2])))
```

with the current code (using `jnp.atleast_2d`) gives `[0.03597286 0.03597286]`. However, if you change line 562 to use `_atleast_2d_consistent` then this example gives `[0.00494525 1.0049453]` instead.

Looking at the type hints involved, and also at line 576 below, I think that `jnp.atleast_2d` is still the correct way to go, but I don't think that this justification is correct.